packages/indexer/src/sync/events/handlers/seaport.ts

```
@@ -209,6 +210,50 @@ export const handleEvents = async (events: EnhancedEvent[], onChainData: OnChain
 
         break;
       }
+
+      case "seaport-v1.2-order-validated":
+      case "seaport-order-validated": {
+        const parsedLog = eventData.abi.parseLog(log);
+        const orderId = parsedLog.args["orderHash"].toLowerCase();
+
+        const tx = await utils.fetchTransaction(baseEventParams.txHash);
+        const orderKind = subKind.startsWith("seaport-v1.2") ? "seaport-v1.2" : "seaport";
+
+        const isV12 = orderKind === "seaport-v1.2";
+        const exchange = isV12
+          ? new Sdk.SeaportV12.Exchange(config.chainId)
+          : new Sdk.Seaport.Exchange(config.chainId);
+        const inoutData = exchange.contract.interface.decodeFunctionData("validate", tx.data);
+        const inputOrders = inoutData.orders;
+        for (let index = 0; index < inputOrders.length; index++) {
+          const orderData = inputOrders[index];
+          try {
+            const { parameters } = orderData;
+            const counter = await exchange.getCounter(baseProvider, parameters.offerer);
+            const order = new Sdk.Seaport.Order(config.chainId, {
+              kind: "single-token",
```

We shouldn't assume an order is `single-token`. Let's just remove the `kind` and make sure to wrap the order creation in a `try / catch` and ignore any errors.